### PR TITLE
FUll editing pass for Spring Data MongoDB

### PR DESCRIPTION
**Pull request by Buzzardo**
I did a full editing pass of the Spring Data MongoDB reference guide. I also adjusted index.adoc to work with the changes I made to the build project, so that we get Epub and PDF as well as HTML.

### CODE REVIEW

File: src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -1,50 +1,48 @@
...
 (C) 2008-2018 The original authors.
 
-NOTE: _Copies of this document may be made for your own use and for distribution to others, provided that you do not charge any fee for such copies and further provided that each copy contains this Copyright Notice, whether distributed in print or electronically._
-
-toc::[]
+NOTE: Copies of this document may be made for your own use and for distribution to others, provided that you do not charge any fee for such copies and further provided that each copy contains this Copyright Notice, whether distributed in print or electronically.
 
 include::preface.adoc[]
 
-:leveloffset: +1
```
Comment (mp911de) [inline src/main/asciidoc/index.adoc]: What's the reason behind this change? We're using `:leveloffset:` in the other docs, too and didn't change these during editing?

**Comment by mp911de** [comment]
Thanks a lot. That's merged, polished, and backported now.

File: src/main/asciidoc/index.adoc (hunk 1 above)
Comment (Buzzardo) [reply src/main/asciidoc/index.adoc]: I discovered that epub generation through Asciidoctor has trouble with doing leveloffset that way but works if I put leveloffset=+1 in each include statement.

Asciidoctor's epub maker is still in alpha, so it has some limitations.
